### ISSUE TITLE
OSD Overlay create a MP4 clip much longer than the original

### ISSUE THREAD

**Issue by slup67** [fpv-wtf/wtfos-configurator#392]
### Device

Goggles V2

### Operating System (Including version)

Windows 11

### Browser

Chrome

### Trace ID (bottom left of the configurator screen)

75e9a36d-b0ee-4833-a697-ca1981bb4aee

### Describe the issue

When the MP4 is rendered & played in VLC media player it runs correctly and ends at 6min & 21sec which is correct. But when mouse hovering over the file the length is reported to be 22min & 54sec.

When then trying to place the file in a timeline in Davinci Resolve the file is reported to be "Offline" & the time line is 22min & 54sec... this means that no editing can be done.

Something appears to go wrong when the rendered OSD-MP4 gets closed.

Attaching the exif info for the file from ExifTool where the track length is reported to be 22min & 55sec.

[DJIG0000-osd.txt](https://github.com/fpv-wtf/wtfos-configurator/files/11962311/DJIG0000-osd.txt)

**Comment by stylesuxx** [comment]
Is this reproducible on your end every time you render the file, no matter the file, or this this happen with only a certain recording?

**Comment by slup67** [comment]
Well... don't overlay the OSD on the goggle video that often & even more rarely want to take the osd overlay video through Davinci Resolve (do it just to upscale it to 4K to be able to use a higher bitrate when uploading to YT).

Just recently saw this (a couple weeks ago) as a final overlayed from the configurator was over an hour long (real length around 15min) when I opened it in DR (DR didn't say "media offline that time)...  after the real end of the video the last frame just repeated for another 45min. had to make a cut in the timeline & throw away the excess. Thought this was odd & checked the  original osd rendered video & exiftools say total length is over an hour & so does windows when mouse hovering over the file. But as said... VLC, Potplayer & MPC-HC player play the original osd vido fine with it's correct length.

This happen on all goggle videos from both my quads with WTFOS Vistas & Goggle V2 (all updated to the latest with wtfos system 0.25)... the overlay configurator works fine, the rendered file plays fine but DR gets upset when trying to bring it into the timeline there & the exif info says that the video is much longer than it should be.

**Comment by slup67** [comment]
Found another original goggle clip + osd rendered I had left since december 2022... original is 37s 950ms long according to mouse hovering in windows but the osd rendered is reported to be 2min & 16s. This overlay render was made on the older site https://knifa-develop.fpv.wtf/osd-overlay

**Comment by FinalFrag** [comment]
@slup67 could you share the 37s clip and corresponding .osd file?

**Comment by slup67** [comment]
Yeah, sure. Original goggle clip+osd file & resulting file from the configurator.

I'm putting the files on my Box cloud storage... files to big to just attach.
https://app.box.com/s/ykbu81k5e65m8pm6l0f34siesgwp3jjr

**Comment by FinalFrag** [comment]
I think I have a solution locally, but I don't have davinci resolve installed, could you try dropping this file into it? It's giving me a weird 'truncated data' error in exiftools, but every viewer I have seems to play it ok.

https://www.dropbox.com/s/vz8g8wsniszheca/DJIG0070-osd.mp4?dl=1

**Comment by slup67** [comment]
The osd rendered clip I got from you (renamed it to DJIG00701-osd.mp4 to be able to handle both clips in the same time line in DR) seems to have the correct length... both in all media players & Davinci Resolve.

As you see in the attached pic, the top clip is from you & below is my own osd rendered clip that I shared with you... which is much longer.

My own clip works in DR without reporting an error about "Offline media"... but this DR error seems to be intermittent. What's consistent though is that the osd rendered clips through the configurator always are too long. So in this sense your "fixed" clip works... but I also see that warning about truncated data in exiftool, but don't know what it means. 

![Davinci Resolve](https://github.com/fpv-wtf/wtfos-configurator/assets/117948411/337fb493-7114-44be-8986-5f1b0d626167)
